addons/web_editor/static/lib/odoo-editor/src/utils/utils.js

```diff
@@ -25,9 +25,9 @@ export const CTGROUPS = {
 };
 
 export const URL_REGEX =
-    /((?:(?:https?:\/\/)|(?:[-a-zA-Z0-9@:%._\+~#=]{1,64}\.))[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,63}\b(?:[^\s]*))/gi;
+    /((?:(?:https?:\/\/)|(?:[-a-zA-Z0-9@:%._\+~#=]{1,64}\.))[-a-zA-Z0-9@:%._\+~#=]{3,256}\.[a-zA-Z0-9()]{2,63}\b(?:(?!\.)[^\s]*))/gi;
 export const URL_REGEX_WITH_INFOS =
-    /((?:(https?:\/\/)|(?:[-a-zA-Z0-9@:%._\+~#=]{1,64}\.))[-a-zA-Z0-9@:%._\+~#=]{1,256}\.[a-zA-Z0-9()]{1,63}\b(?:[^\s]*))/gi;
+    /((?:(https?:\/\/)|(?:[-a-zA-Z0-9@:%._\+~#=]{1,64}\.))[-a-zA-Z0-9@:%._\+~#=]{3,256}\.[a-zA-Z0-9()]{2,63}\b(?:(?!\.)[^\s]*))/gi;
```

Comment:
`(?![\s\.])` instead of `(?!\.)[^\s]` would be clearer imo instead of mixing two syntaxes for the same thing.

But this makes that addresses `.co.uk` for instance aren't recognized as  url, right?
@sebgeelen since this was your regex originally.

Reply:
`amazon.co.uk` is not recognized anymore because the `co` is considered as the main domain, and it cannot be less than 3 char in the changed regex.

So it's a regression IMO

Reply:
> `(?![\s\.])` instead of `(?!\.)[^\s]` would be clearer imo instead of mixing two syntaxes for the same thing.

The regex you propose will have a different effect.

Mine is "any character that is not a space (including dot) but the dot cannot be directly after something like foo.bar.tld.x"
It solves the cases that without it, `foo.bar.tld.x` would be considered a good match but shouldn't. 

I don't understand yours `(?![\s\.])`, the `\s` is guaranteed by the next  `[^\s]*`

Keep in mind that `(?!)` is a negative lookahead and do not "consume" character.

Reply:
> Do a domain with two character seems ok with you ?

sure